starter-code/src/functions-and-arrays.js

```diff
@@ -11,14 +17,55 @@ var words = [
   'crackpot'
 ];
 
+function findLongestWord(arr) {
```

Comment:
Esta función te estará fallando en algunos casos:

Fíjate al declarar el bucle for, que haces un `i < arr.length - 1`, si lo dejamos así, no estamos teniendo en cuenta el último elemento del array y no funcionará, tu función refactorizada quedaría así:

```
function findLongestWord(arr) {
  var longest = "";
  if (arr.length <= 0) {
    return undefined;
  }
  for (var i = 0; i < arr.length; i++) {
    if (arr[i].length > longest.length) {
      longest = arr[i];
    }
  }
  return longest;
}
```